io-functions-pay-portal/ActivatePayment/__tests__/handler.test.ts

```diff
@@ -1,11 +1,81 @@
 /* tslint:disable: no-any */
 
-import { HttpHandler } from "../handler";
-
-describe("HttpCtrl", () => {
-  it("should return a string when the query parameter is provided", async () => {
-    const httpHandler = HttpHandler();
-    const response = await httpHandler();
-    expect(response.kind).toBe("IResponseSuccessJson");
-  });
+import { Context } from "@azure/functions";
+import { right } from "fp-ts/lib/Either";
+import { PaymentActivationsPostRequest } from "../../generated/pagopa-proxy/PaymentActivationsPostRequest";
+import { ActivatePaymentHandler } from "../handler";
+
+export const context = ({
+  bindings: {},
+  log: {
+    // tslint:disable-next-line: no-console
+    error: jest.fn().mockImplementation(console.log),
+    // tslint:disable-next-line: no-console
+    info: jest.fn().mockImplementation(console.log),
+    // tslint:disable-next-line: no-console
+    verbose: jest.fn().mockImplementation(console.log),
+    // tslint:disable-next-line: no-console
+    warn: jest.fn().mockImplementation(console.log)
+  }
+} as any) as Context;
+
+afterEach(() => {
+  jest.resetAllMocks();
+  jest.restoreAllMocks();
+});
+
+const validPaymentActivationsRequest = {
```

Comment:
Not a critical point, just an idea: wouldn't it be nice to collect all the _fake_ data in a dedicated module so as to allow random generation of test inputs or _fuzz testing_?

Reply:
yes